python/cudf/cudf/core/column/numerical_base.py

```diff
@@ -171,6 +171,12 @@ def quantile(
                 result = type(self).from_pylibcudf(plc_column)  # type: ignore[assignment]
         if return_scalar:
             scalar_result = result.element_indexing(0)
+            if cudf.get_option("mode.pandas_compatible") and scalar_result in {
+                None,
+                NA,
+                np.nan,
+            }:
+                return _get_nan_for_dtype(self.dtype)
```

Comment:
Since `element_indexing` should return `_PANDAS_NA_VALUE` for null-like results, what is this additional case addressing?

Reply:
Yup, this is now redundant. Removed this.